Added Open-Ended Evaluation

niuyizhan opened the pull request.

niuyizhan committed:
Added Open-Ended Evaluation

not implemented hallucination evaluation for open-ended evaluation yet

ChaoPang commented on evaluations/qa/medquad_evaluator.py:
```diff
@@ -0,0 +1,72 @@
+from evaluations.causal_llm_evaluators import CausalLanguageModelEvaluator
+from prompt_templates.qa.qa_prompt import MedQuADQuestionAnswerPromptV1, MedQuADQuestionAnswerPromptBaseV1
+
+
```
I am not sure if you need this custom metric since you are using the standard rogue, bleu and meteor

ChaoPang commented on evaluations/qa/medquad_evaluator.py:
```diff
@@ -0,0 +1,72 @@
+from prompt_templates.qa.qa_prompt import MedQuADQuestionAnswerPromptV1, MedQuADQuestionAnswerPromptBaseV1
+
+
+class PointWiseScore:
+    @staticmethod
+    def compute(
+            predictions,
+            references,
+            **kwargs
+    ):
+        match_score = np.sum((predictions == references).astype(int) * 1)
+        mismatch_score = - np.sum((predictions != references).astype(int) * 0.25)
+        return match_score + mismatch_score
```
I am not sure if you need this custom metric since you are using the standard rogue, bleu and meteor

```suggestion
```

ChaoPang commented on evaluations/hallucination/medquad_hallucination_evaluator.py:
```diff
@@ -0,0 +1,59 @@
+        match_score = np.sum((predictions == references).astype(int) * 1)
+        mismatch_score = - np.sum((predictions != references).astype(int) * 0.25)
+        normalized_score = (match_score + mismatch_score) / len(predictions)
+        return {'point_wise_score': normalized_score}
```
Similarly, I don't think `PointWiseScore` is used anywhere.
```Suggestion
```

ChaoPang commented on prompt_templates/qa/qa_prompt.py:
```diff
@@ -94,3 +96,56 @@ def extract_answer(
     @staticmethod
     def map_answer(answer):
         return PubmedQuestionAnswerPromptJsonV1.answer_idx_mapping.get(answer.lower().strip(), 3)
+
```
Could you move the MedQuAD prompts to a separate module?

ChaoPang reviewed (changes requested):
I don't think the runner is included in this PR.

niuyizhan committed:
Update evaluations/qa/medquad_evaluator.py

Co-authored-by: Chao Pang <ChaoPang229@gmail.com>

niuyizhan committed:
Update evaluations/hallucination/medquad_hallucination_evaluator.py

Co-authored-by: Chao Pang <ChaoPang229@gmail.com>

niuyizhan committed:
add_open_ended_runner

niuyizhan committed:
changed closed_book_qa to a different module

niuyizhan committed:
change the prompt import path

niuyizhan commented:
I have updated the runner file to the niuyizhan-patch-1 branch, changed closed_book_qa_prompt to a separate file, and changed medquad_evaluator file correspondingly!

ChaoPang commented:
@niuyizhan I don't think you need a separate evaluator for hallucination because the prompt is the same as the closed-book prompt. You could add two additional hallucination metrics to the MedQuAD evaluator. I have a separate evaluator for PubmedQA because the pubmedqa hallucination prompt is designed to solicit hallucinated content. I will accept the PR, please open a new PR to remove `medquad_hallucination_evaluator` and moved its two hallucination metrics to `medquad_evaluator`

niuyizhan commented:
Sure thing, thanks for the advice Chao.
